src/widgetastic_patternfly/__init__.py

```diff
@@ -2351,3 +2359,219 @@ def item_select(self, item, close=True):
+        """read all data on chart
+
+        Returns:
+            :py:class:`list` complete data on chart
+        """
+        data = []
+        for el in self.browser.elements(self.RECT):
+            self.browser.move_to_element(el)
+            data.append(self.tooltip.read())
+        return data
+
+
+class SingleLineChart(SparkLineChart):
+    """Represents the Single Line Chart from Patternfly (Data Visualization).
+
+    Args:
+        id: id of SingleLineChart
+        locator: If id not applies, you can also supply a full locator
+
+    .. code-block:: python
+        single_line_chart = SingleLineChart(id="singleLineChart")
+    """
+    X_AXIS = ".//*[contains(@class, 'c3-axis c3-axis-x')]/*[contains(@class, 'tick')]"
+    tooltip = Table(locator=".//div[contains(@class,'c3-tooltip-container')]/table")
+
+    @property
+    def _elements(self):
+        br = self.browser
+        return {
+            x.get_attribute("textContent"): el
+            for (x, el) in zip(br.elements(self.X_AXIS), br.elements(self.RECT))
+        }
+
+    def _get_data(self, elements):
+        data = {}
+        for el in elements:
+            self.tooltip.clear_cache()
+            self.browser.move_to_element(el)
+            raw_data = {row[0].text: row[1].text for row in self.tooltip.rows()}
+            if self.tooltip.headers:
+                tooltip_data = {self.tooltip.headers[0]: raw_data}
+            else:
+                # In the absence of a header, `:` separates x-axis points and values.
+                tooltip_data = {
+                    h[:-1] if (h[-1] == ":") else h: value for h, value in raw_data.items()
+                }
+            data.update(tooltip_data)
+        return data
+
+    def read(self):
+        """read all data on chart
+
+        Returns:
+            :py:class:`dict` complete data on chart
+        """
+        return self._get_data(self._elements.values())
+
+    def get_values(self, x_axis):
+        """data for specific x-axis point on chart
+
+        Args:
+            x_axis: x-axis point as per chart
+
+        Returns:
+            :py:class:`dict` data for selected timestamp
+        """
+        el = [self._elements.get(x_axis)]
+        return self._get_data(el)
+
+
+class LineChart(SingleLineChart):
+    """Represents the Line Chart having legends from Patternfly (Data Visualization).
+
+    Args:
+        id: id of LineChart
+        locator: If id not applies, you can also supply a full locator
+
+    .. code-block:: python
+        line_chart = LineChart(id="lineChart")
+    """
+    LEGENDS = ".//*[contains(@class, 'c3-legend-item c3-legend-item-')]"
+
+    @property
+    def _legends(self):
+        return {self.browser.text(leg): leg for leg in self.browser.elements(self.LEGENDS)}
+
+    @property
+    def legends(self):
+        """ Get all available legends
+
+        Returns:
+            :py:class:`list` all available legends
+        """
+        return list(self._legends.keys())
```

Comment:
Why is it necessary to do `list(self._legends.keys())` here? Isn't `self._legends.keys()` already a list?

Reply:
Also, why only have a `property` for the text of each legend? What about for the browser elements? e.g.
```
@property 
def legend_text():
    ...
@property
def legend_element():
    ...
```
This may be more readable, but not really necessary.

Reply:
May be @RonnyPfannschmidt can give me some hack and learning approach towards such handling.

Reply:
on python3 dict keys are indeed a iterables instead of a materialized list
the multi python version support pattern is to indeed wrap keys into a list call

depending on the intended usage-patterns it may beneficial to return a `set(somedict)` instead of a `list(somedict)` - what are those results used for?

Reply:
those are `legends` means `str` elements for graph data pointing things.